src/server/services/procedures/earth-orbit/earth-orbit.js

```diff
@@ -0,0 +1,192 @@
+ * http://vo.imcce.fr/insola/earth/online/earth/earth.html
+ *
+ * Original datasets are available at
+ * http://vo.imcce.fr/insola/earth/online/earth/La2004/INSOLN.LA2004.BTL.100.ASC
+ * http://vo.imcce.fr/insola/earth/online/earth/La2004/INSOLN.LA2004.BTL.250.ASC
+ * http://vo.imcce.fr/insola/earth/online/earth/La2004/INSOLN.LA2004.BTL.ASC
+ * http://vo.imcce.fr/insola/earth/online/earth/La2004/INSOLP.LA2004.BTL.ASC
+ * http://vo.imcce.fr/webservices/miriade/proxy.php?file=http://145.238.217.35//tmp/insola/insolaouto7Yk3u&format=text
+ * http://vo.imcce.fr/webservices/miriade/proxy.php?file=http://145.238.217.38//tmp/insola/insolaouteXT96X&format=text
+ * @alpha
+ * @service
+ * @category Science
+ * @category Climate
+ */
+const fs = require('fs');
+const path = require('path');
+
+const EarthOrbit = {};
+
+// this contains eccentricity, obliquity, and longitude data
+EarthOrbit._dataEccOblLongi = [];
+
+// this contains insolation data 
+EarthOrbit._dataInsolation = [];
+
+// the contains precession data
+EarthOrbit._dataPrecession = [];
+
+// converting time to year CE, note: the present in here is the year 2000
+// because it is not using BP present (1950)
+let timeConversion = function(time) {
+    return (2000 + 1000*time);
+};
+
+// Initialize startyear to -1000000 CE and endyear to 1000000 CE if no inputs are given.
+let checkInputYear = function(startyear, endyear) {
+    if(startyear == null) {
+        startyear = -1000000;
+    }
+    if(endyear == null) {
+        endyear = 1000000;
+    }
+    return [startyear, endyear];
+};
+
+// Normalize all time values to decimal numbers so that they can be used for computations. 
+// For example, converting 10506D-05 to 0.10506
+let parseDecimalValue = function(value) {
+    if(value.charAt(value.length-4)=='D') {
+        return Number(value.substring(0,value.length-4)) * Math.pow(10,Number(value.substring(value.length-3,value.length)));
+    } else {
+        return value;
+    }
+};
+
+// reading and importing obliquity, eccentricity, and longitude data
+const importData2004 = function(line) {
+    let [year, eccentricity, obliquity, longitude] = line.trim().split(/\s+/);
+    if(year.charAt(year.length-1)=='.') {
+        year = timeConversion(Number(year.substring(0,year.length-1)));
+    } else {
+        year = timeConversion(year);
+    }
+    eccentricity = parseDecimalValue(eccentricity);
+    obliquity = parseDecimalValue(obliquity);
+    longitude = parseDecimalValue(longitude);
+    EarthOrbit._dataEccOblLongi.push({year, eccentricity, obliquity, longitude});
+};
+
+// reading and importing insolation data
+const importDataInsol = function(line) {
+    let [year, value] = line.trim().split(/\s+/);
+    year = timeConversion(year);
+    EarthOrbit._dataInsolation.push({year, value});
+};
+
+// reading and importing precession data
+const importDataPrec = function(line) {
+    let [year, value] = line.trim().split(/\s+/);
+    year = timeConversion(year);
```

Comment:
Like the comment on the IceSheets, it would be great to do the parsing and validating here.